Atomic/syscontainers.py

```diff
@@ -1550,6 +1557,32 @@ def prune_ostree_images(self):
         refs = {}
         app_refs = []
 
+        # Check for dangling checkouts that couldn't be deleted on uninstall.
+        checkouts = self._get_system_checkout_path()
+        for i in os.listdir(checkouts):
+            if i[0] == '.':
+                continue
+            if i[-2:] not in [".0", ".1"]:
+                continue
+
+            container = i[:-2]
+
+            # Check if the container is still installed.
+            if os.path.lexists(os.path.join(checkouts, container)):
+                continue
+
+            fullpath = os.path.join(checkouts, i)
+
+            if not os.path.isdir(fullpath):
+                continue
+
+            # If we got here, the deployment is not related to an installed
+            # container, so delete it.
+            try:
+                shutil.rmtree(fullpath)
+            except OSError:
+                pass
```

Comment:
A log line here may be good noting the error.

Reply:
added a fixup patch